.github/workflows/nightly-test.yml

```diff
@@ -19,6 +19,7 @@ jobs:
           - os: macos-latest # ASAN build
             sanitizer: "address"
             assertions: ON
+            build_type: "Release"
```

Comment:
Is this compatible with address sanitizer?
Does it need debug info?
I wonder if we can do `RelWithDebInfo`